File: Programmers/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        for (int i = 1; i <= yellow; i++) {
            if (yellow % i != 0) continue;

            int row = i;
            int col = yellow / row;
            if (row > col) break;

            if ((col + 2) * 2 + (row * 2) == brown) {
                answer[0] = col + 2;
                answer[1] = row + 2;
            }
        }
        return answer;
    }
}

File: Programmers_js/카펫.js
function solution(brown, yellow) {
    let answer = [];
    
    for(let i=1; i<=yellow; i++) {
        let w = i;
        let h = yellow / w;
        
        if(w < h) {
            continue;
        }
        
        if((w+2)*(h+2) - yellow === brown) {
            answer = [w+2, h+2];
            break;
        }
    }
    return answer;
}
